WalletWasabi.Tests/RegTests.cs

```diff
@@ -1135,6 +1136,133 @@ private void WalletTestsAsync_MemPoolService_TransactionReceived(object sender,
+			var indexDownloader = new IndexDownloader(Global.RpcClient.Network, indexFilePath, new Uri(RegTestFixture.BackendEndPoint));
+
+			// 4. Create key manager service.
+			var keyManager = KeyManager.CreateNew(out Mnemonic mnemonic, "password");
+
+			// 5. Create wallet service.
+			var blocksFolderPath = Path.Combine(SharedFixture.DataDir, nameof(SendTestsFromHiddenWalletAsync), $"Blocks");
+			var wallet = new WalletService(keyManager, indexDownloader, memPoolService, nodes, blocksFolderPath);
+			wallet.NewFilterProcessed += Wallet_NewFilterProcessed;
+
+			Assert.Empty(wallet.Coins);
+
+			// Generate script
+			var scp = new Key().ScriptPubKey;
+			
+			// Get some money, make it confirm.
+			var key = wallet.GetReceiveKey("foo label");
+			var txid = await Global.RpcClient.SendToAddressAsync(key.GetP2wpkhAddress(network), new Money(1m, MoneyUnit.BTC));
+
+			// Generate some coins
+			await Global.RpcClient.GenerateAsync(2);
+
+			try
+			{
+				nodes.Connect(); // Start connection service.
+				node.VersionHandshake(); // Start mempool service.
+				indexDownloader.Synchronize(requestInterval: TimeSpan.FromSeconds(3)); // Start index downloader service.
+
+				// Wait until the filter our previous transaction is present.
+				var blockCount = await Global.RpcClient.GetBlockCountAsync();
+				await WaitForFiltersToBeProcessedAsync(TimeSpan.FromSeconds(120), blockCount);
+
+				using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30)))
+				{
+					await wallet.InitializeAsync(cts.Token); // Initialize wallet service.
+				}
+				Assert.Single(wallet.Coins);
+
+				// Send money before reorg.
+				var operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				var btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				// Test synchronization after fork.
+				// Invalidate the blocks containing the funding transaction
+				var tip = await Global.RpcClient.GetBestBlockHashAsync();
+				await Global.RpcClient.InvalidateBlockAsync(tip); // Reorg 1
+				tip = await Global.RpcClient.GetBestBlockHashAsync();
+				await Global.RpcClient.InvalidateBlockAsync(tip); // Reorg 2
+
+				// Generate two new blocks (replace the previous invalidated ones)
+				_filtersProcessedByWalletCount = 0;
+				await Global.RpcClient.GenerateAsync(3);
+				await WaitForFiltersToBeProcessedAsync(TimeSpan.FromSeconds(120), 3);
+
+				// Send money after reorg.
+				// When we invalidate a block, those transactions setted in the invalidated block
+				// are reintroduced when we generate a new block though the rpc call 
+				operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				btx = await wallet.BuildTransactionAsync("password", operations, 2);
```

Comment:
In the comments you say you send money, but you don't you just building the transactions. If you intend to also propagate them do it with `wallet.SendTransactionAsync`